x/metadata/keeper/keeper_test.go

```diff
@@ -460,6 +592,106 @@ func (s *KeeperTestSuite) TestValidateAllOwnersAreSigners() {
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              fmt.Sprintf("missing signature from existing owner %s; required for update", s.user3),
+		},
+		{
+			name:                  "Scope Spec with 3 owners - one signer with a special case message type - with grant - authz",
+			owners:                []string{s.user1, s.user2, s.user3},
+			signers:               []string{s.user1, s.user3}, // signer 3 is grantee of singer 2
+			msgTypeURL:            types.TypeURLMsgAddScopeDataAccessRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name:                  "Scope Spec with 3 owners - two signers with a special case message type - grant on parent of special case message type - authz",
+			owners:                []string{s.user1, s.user2, s.user3},
+			signers:               []string{s.user1, s.user3}, // signer 3 grantee of signer 2
+			msgTypeURL:            types.TypeURLMsgDeleteContractSpecFromScopeSpecRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name:                  "Scope Spec with 2 owners - one signer - no grant - authz - error",
+			owners:                []string{s.user2, s.user3},
+			signers:               []string{s.user3},
+			msgTypeURL:            types.TypeURLMsgDeleteRecordRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               nil,
+			grantee:               nil,
+			errorMsg:              fmt.Sprintf("missing signature from existing owner %s; required for update", s.user2),
+		},
+	}
+
+	now := s.ctx.BlockHeader().Time
+	s.Require().NotNil(now)
+
+	// Test cases
+	for _, tc := range tests {
+		s.T().Run(tc.name, func(t *testing.T) {
+			a := authz.NewCountAuthorization(tc.msgTypeURL, tc.allowedAuthorizations)
+			err := s.app.AuthzKeeper.SaveGrant(s.ctx, tc.grantee, tc.granter, a, now.Add(time.Hour))
+			require.NoError(t, err)
+
+			err = s.app.MetadataKeeper.ValidateAllOwnersAreSignersWithAuthz(s.ctx, tc.owners, tc.signers, tc.msgTypeURL)
+			if len(tc.errorMsg) == 0 {
+				assert.NoError(t, err, "ValidateAllOwnersAreSigners unexpected error", tc.name)
+			} else {
+				assert.EqualError(t, err, tc.errorMsg, "ValidateAllOwnersAreSigners error", tc.name)
+			}
+
+			// validate allowedAuthorizations
+			if err == nil {
+				auth, _ := s.app.AuthzKeeper.GetCleanAuthorization(s.ctx, tc.grantee, tc.granter, tc.msgTypeURL)
+				if tc.allowedAuthorizations == 1 {
+					// authorization is deleted after one use
+					assert.Nil(t, auth)
+				} else {
+					actual := auth.(*authz.CountAuthorization).AllowedAuthorizations
+					assert.Equal(t, tc.allowedAuthorizations-1, actual)
+				}
+			}
+		})
+	}
+}
+
```

Comment:
I'd like to see a test with two owners (1 & 2), and two different signers (3 & 4), with two authz count authorization grants: granter: 1, grantee: 3, count: 1; and granter: 2, grantee: 4, count: 2. Then a message is executed and we check that both grants have updated appropriately. Then another message is executed and we make sure it fails (because the first grant only had 1 use), and that the second grant still exists with 1 use left.